File: day1/day1.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

class Day1{
    private:
        vector<int> input;

    public:
        Day1(string input_name){
            ifstream infile(input_name);
            string line;
            while (getline(infile, line)){
                input.push_back(stoi(line));
            };
        };

        int sumOfFuel(){
            int fuel = 0;
            for (auto i: input){
                fuel += i / 3 -2;
            }

            return fuel;
        }

        int sumOfFuelAdded(){
            int fuel = 0;
            for (auto i: input){
                int added_fuel = i;
                while (added_fuel){
                    added_fuel = max(added_fuel / 3 -2, 0);
                    fuel += added_fuel;
                }
            }
            return fuel;

        }
        
};

int main(){
    
    Day1 day1 = Day1("input.txt");
    cout<< day1.sumOfFuel() << endl;
    cout<< day1.sumOfFuelAdded() << endl;

    return 0;
}

File: day1/day1.py
class Day1():
    def __init__(self):
        pass 
    
    def sum_of_fuel(self, inputs):
        fuel = 0
        for mass in inputs:
            fuel += int(mass) // 3 -2
        return fuel

    def sum_of_fuel_added(self,inputs):
        fuel = 0 
        for mass in inputs:
            added_fuel = int(mass)
            while added_fuel:
                added_fuel = max(added_fuel // 3 - 2, 0)
                fuel += added_fuel

        return fuel

if __name__ == "__main__":

    with open('input.txt', 'r') as r:
        content = r.readlines()
    
    print(Day1().sum_of_fuel(content))
    print(Day1().sum_of_fuel_added(content))

